sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/ManagementPolicyImpl.java

```diff
@@ -131,55 +130,11 @@ public String type() {
     public List<PolicyRule> rules() {
         List<ManagementPolicyRule> originalRules = this.policy().rules();
-        List<PolicyRule> returnRules = new ArrayList<>();
-        for (ManagementPolicyRule originalRule : originalRules) {
-            List<String> originalBlobTypes = originalRule.definition().filters().blobTypes();
-            List<BlobTypes> returnBlobTypes = new ArrayList<>();
-            for (String originalBlobType : originalBlobTypes) {
-                returnBlobTypes.add(BlobTypes.fromString(originalBlobType));
-            }
-            PolicyRule returnRule =
-                new PolicyRuleImpl(originalRule.name())
-                    .withLifecycleRuleType()
-                    .withBlobTypesToFilterFor(returnBlobTypes);
-
-            // building up prefixes to filter on
-            if (originalRule.definition().filters().prefixMatch() != null) {
-                ((PolicyRuleImpl) returnRule)
-                    .withPrefixesToFilterFor(originalRule.definition().filters().prefixMatch());
-            }
-
-            // building up actions on base blob
-            ManagementPolicyBaseBlob originalBaseBlobActions = originalRule.definition().actions().baseBlob();
-            if (originalBaseBlobActions != null) {
-                if (originalBaseBlobActions.tierToCool() != null) {
-                    ((PolicyRuleImpl) returnRule)
-                        .withTierToCoolActionOnBaseBlob(
-                            originalBaseBlobActions.tierToCool().daysAfterModificationGreaterThan());
-                }
-                if (originalBaseBlobActions.tierToArchive() != null) {
-                    ((PolicyRuleImpl) returnRule)
-                        .withTierToArchiveActionOnBaseBlob(
-                            originalBaseBlobActions.tierToArchive().daysAfterModificationGreaterThan());
-                }
-                if (originalBaseBlobActions.delete() != null) {
-                    ((PolicyRuleImpl) returnRule)
-                        .withDeleteActionOnBaseBlob(
-                            originalBaseBlobActions.delete().daysAfterModificationGreaterThan());
-                }
-            }
-
-            // build up actions on snapshot
-            ManagementPolicySnapShot originalSnapshotActions = originalRule.definition().actions().snapshot();
-            if (originalSnapshotActions != null) {
-                if (originalSnapshotActions.delete() != null) {
-                    ((PolicyRuleImpl) returnRule)
-                        .withDeleteActionOnSnapShot(originalSnapshotActions.delete().daysAfterCreationGreaterThan());
-                }
-            }
-            returnRules.add(returnRule);
-        }
-        return Collections.unmodifiableList(returnRules);
+        return originalRules == null
+            ? Collections.emptyList()
+            : originalRules.stream()
+            .map(rule -> new PolicyRuleImpl(rule, this))
+            .collect(Collectors.toList());
```

Comment:
I guess originally this clone of rule is intended for read-only purpose? 

However, this adds to the risk of not supporting future added action types. I saw similar structure of rules in `ApplicationGateway`, and it doesn't use clone as return.